File: 1930-unique-length-3-palindromic-subsequences/1930-unique-length-3-palindromic-subsequences.cpp
class Solution {
public:
    int countPalindromicSubsequence(string s) {
        char c[26] = {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'};
        int a = 0, t = 0;

        int l, r;
        for (char x : c) {
            for (l = 0; l < s.length() && s[l] != x; l++);
            if (l == s.length()) continue;
            for (r = s.length() - 1; r >= 0 && s[r] != x; r--);
            if (l >= r) continue;

            vector<bool> v(128, false);
            t = 0;
            for (int i = l + 1; i < r; i++) {
                if (!v[s[i]]) {
                    v[s[i]] = true;
                    t++;
                    if (t == 26) break;
                }
            }
            a += t;
        }
        return a;
    }
};

File: 1930-unique-length-3-palindromic-subsequences/1930-unique-length-3-palindromic-subsequences.py
class Solution:
    def countPalindromicSubsequence(self, s: str) -> int:
        dic={}
        for i in range(0,len(s)):
            if s[i] in dic:
                dic[s[i]][1]=i
            else:
                dic[s[i]]=[i,0]
        count=0
        for k,value in dic.items():
            ss=set()
            for i in range(value[0]+1,value[1]):
                ss.add(s[i])
            count=count+len(ss)
        return count